src/main/java/org/ice4j/ice/Component.java

```diff
@@ -465,7 +463,24 @@ public void updateRemoteCandidates()
             {
                 for (CandidatePair pair : checkList)
                 {
-                    streamCheckList.add(pair);
+                    /* Check whether the pair is already in the check list.
+                     * (This can happen for pairs with remote peer-reflexive candidates,
+                     *  since those candidates aren't added to the candidate list even
+                     *  though the pair is added to the check list).
+                     */
+                    CandidatePair existingPair = streamCheckList.findPairMatching(pair.getLocalCandidate(), pair.getRemoteCandidate());
+                    if (existingPair != null) {
+                        logger.info("existing Pair updated: " +
+                            existingPair.toShortString() +
+                            " to " + pair.toShortString() + ".");
+                        existingPair.setRemoteCandidate(pair.getRemoteCandidate());
```

Comment:
Should we be smarter about which remote candidate we maintain? I.e. make sure we use whichever one is not prflx? Conceivably if candidates trickle and are delayed in signaling we may get a prflx prior to a signaled one.

Reply:
Ones that come in signaling should always win over ones that were discovered as prflx (this can be determined through deep textual analysis of RFC 5245).

The only way we hit this code is if the check being replaced is on the checklist but its remote candidate isn't on the remote candidate list, i.e. if this check was indeed discovered as prflx.